libraries/AP_HAL_SITL/UARTDriver.cpp

```diff
@@ -284,6 +287,8 @@ size_t UARTDriver::_write(const uint8_t *buffer, size_t size)
         }
 #endif // HAL_BUILD_AP_PERIPH
 
+    // Include lost byte in tx count, we think we sent it even though it was never added to the write buffer
+    _tx_stats_bytes += lost_byte;
```

Comment:
Strictly the lost bytes should be included in the baud limit calculation. Currently there not, so the baud limit only applies to the bytes that are not lost. So loosing makes no difference to the total possible data rate.